Stripe/STPRedirectContext.m

```diff
@@ -138,18 +149,21 @@ - (void)subscribeToUrlAndForegroundNotifications {
 }
 
 - (void)unsubscribeFromNotificationsAndDismissPresentedViewControllers {
+    [self unsubscribeFromNotifications];
+    [self dimissPresentedViewController];
+}
+
+- (void)unsubscribeFromNotifications {
     [[NSNotificationCenter defaultCenter] removeObserver:self
                                                     name:UIApplicationWillEnterForegroundNotification
                                                   object:nil];
     [[STPURLCallbackHandler shared] unregisterListener:self];
+}
 
+- (void)dimissPresentedViewController {
     if (self.safariVC) {
         [self.safariVC.presentingViewController dismissViewControllerAnimated:YES
-                                                                   completion:^{
-                                                                       stpDispatchToMainThreadIfNecessary(^{
-                                                                           [self handleRedirectCompletionWithError:nil];
-                                                                       });
-                                                                   }];
+                                                                   completion:nil];
```

Comment:
We never needed it afaict, it was just an oversight. The only ways this can get called is: 
1. Via through the `handleRedirectCompletionWithError` which calls completion after.
2. You are getting dealloc'd (in which case I'm not sure calling completion makes sense?)
3. You manually called `cancel` on the redirect context (again, in which case I don't think it makes sense to fire the completion)